### PR TITLE
Document how to use different helm credentials for each path

### PR DESCRIPTION
**Pull request by raulcabello**
relates to https://github.com/rancher/fleet/pull/1557

### CODE REVIEW

File: docs/gitrepo-add.md [hunk 1]
```diff
@@ -168,6 +168,39 @@ For example, to add a secret in kubectl, run
...
 After secret is created, specify the secret to `gitRepo.spec.helmSecretName`. Make sure secret is created under the same namespace with gitrepo.
 
+### Use different helm credentials for each path
+
+:::info
+`gitRepo.spec.helmSecretName` will be ignored if `gitRepo.spec.helmSecretNameForPaths` is provided
+:::
+
+Create a file `secrets-path.yaml` that contains credentials for each path defined in a `GitRepo`. Credentials will not be used
+for paths that are not present in this file.
```
Comment (manno) [inline docs/gitrepo-add.md:178]:
```suggestion
The path is the actual path to the bundle within the git repository, which might have more segments than the entry under `paths:`. 
```

**Comment by weyfonk** [comment]
Should the `Danger` paragraph [here](https://fleet.rancher.io/gitrepo-add#using-private-helm-repositories) be edited or replaced with a reference to this new feature?

**Comment by raulcabello** [comment]
> Should the `Danger` paragraph [here](https://fleet.rancher.io/gitrepo-add#using-private-helm-repositories) be edited or replaced with a reference to this new feature?

Good point! done in https://github.com/rancher/fleet-docs/pull/69/commits/f8395dc9a29bc16d77557f0da345b15f818e947f

File: docs/gitrepo-add.md [hunk 2]
```diff
@@ -168,6 +168,39 @@ For example, to add a secret in kubectl, run
 
 After secret is created, specify the secret to `gitRepo.spec.helmSecretName`. Make sure secret is created under the same namespace with gitrepo.
 
+### Use different helm credentials for each path
+
+:::info
+`gitRepo.spec.helmSecretName` will be ignored if `gitRepo.spec.helmSecretNameForPaths` is provided
+:::
+
+Create a file `secrets-path.yaml` that contains credentials for each path defined in a `GitRepo`. Credentials will not be used
```
Comment (weyfonk) [inline docs/gitrepo-add.md:177]: This line ends with what looks like an incomplete sentence; a leftover from an edit maybe? :thinking:

**Review by weyfonk — CHANGES_REQUESTED** [review]

File: docs/gitrepo-add.md (hunk 2 above)
Comment (raulcabello) [reply docs/gitrepo-add.md:177]: ohh good catch! It broke when I applied https://github.com/rancher/fleet-docs/pull/69#discussion_r1206849784 
fixed in https://github.com/rancher/fleet-docs/pull/69/commits/600dd350f51e30ba94e308be934417cd4867d14b

File: docs/gitrepo-add.md [hunk 3]
```diff
@@ -168,6 +168,40 @@ For example, to add a secret in kubectl, run
...
+for paths that are not present in this file.
+The path is the actual path to the bundle within the git repository, which might have more segments than the entry under `paths:`.
```
Comment (weyfonk) [inline docs/gitrepo-add.md:179]: This line reads a bit confusing to me; are readers of this document expected to know what a _bundle_ is in this context, considering that it's a [lower-level concept](https://fleet.rancher.io/bundle-add) than a `GitRepo`? By `the actual path to the bundle`, do we mean a path to a `fleet.yaml`?
Would an example help perhaps?
ie, do we mean that if a `GitRepo` spec contains:
```
spec:
  repo: http://foo.com/bar
  paths:
  - baz/my_path
```
... then `secrets-path.yaml` could look like:
```
baz/my_path/dir_where_fleet_yaml_lives: # more segments than in the `paths` entry above
  username: user
  password: pass
```
?

Happy to discuss, as I'm not yet sure that I understand this.

Comment (raulcabello) [reply docs/gitrepo-add.md:179]: In this case `secrets-path.yaml` could look like
```
baz/my_path: # more segments than in the `paths` entry above
  username: user
  password: pass
```
Fleet will create a Bundle for each dir that has a fleet.yaml. More info https://fleet.rancher.io/gitrepo-content#how-repos-are-scanned

If any subdirectory of `baz/my_path` contains a `fleet.yaml` a new Bundle will be created.   `secrets-path.yaml`  for this subdir would look like:
```
baz/my_path/subdir: # more segments than in the `paths` entry above
  username: user
  password: pass
```

Not sure how to make this clearer, maybe a link to https://fleet.rancher.io/gitrepo-content#how-repos-are-scanned before so users can see how Bundles are created?
Any other suggestion is very welcome

Comment (weyfonk) [reply docs/gitrepo-add.md:179]: Thanks, this is clearer to me; I think we could:
* either add a link to https://fleet.rancher.io/gitrepo-content#how-repos-are-scanned, as you're suggesting
* or simply add a few words to say that the path to a bundle is the path to a `fleet.yaml` file, eg.:
```suggestion
The path is the actual path to the bundle (ie to a folder containing a `fleet.yaml` file) within the git repository, which might have more segments than the entry under `paths:`. 
```
or even:
```suggestion
The path is the actual path to the bundle (ie to a folder containing a `fleet.yaml` file) within the git repository, which might be a subfolder of an entry under `paths:`. 
```

**Review by weyfonk — APPROVED** [review]

**Pull request merged**